src/transformers/models/transfo_xl/modeling_transfo_xl.py

```diff
@@ -1097,16 +1097,30 @@ def forward(
         last_hidden = transformer_outputs[0]
         pred_hid = last_hidden[:, -tgt_len:]
 
+        if labels is not None:
+            # Prevents all labels being -100 and throwing an error
+            # when backwarding the loss
+            miss_valid_label = labels[0, 1:].sum() == (labels.size(1) - 1) * -100
+            if miss_valid_label:
+                # Sets an <EOS> token, just to prevent loss from being NaN
+                labels[0, 1] = self.config.eos_token_id
```

Comment:
Could you give a bit more background on why this is necessary? I don't really see why the exactly those labels should be EOS

Reply:
Since we are dealing with a customizable softmax/loss layer [ProjectiveAdaptiveLogSoftmax](https://github.com/huggingface/transformers/blob/master/src/transformers/models/transfo_xl/modeling_transfo_xl_utilities.py#L29), it is unable to deal when all of the labels are out-of-vocabulary tokens, for example, pad tokens which were replaced with `-100`. Thus, if we feed a label which is fully composed of `-100`, the output loss would be `0`, its average would be `0` and it would throw an error when trying to be backwarded. 

Ideally, would be nice if we could mimic the behavior of the [CrossEntropyLoss](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html), which takes into an account a index that can be ignored (`ignore_index=-100`), and is internally able to deal whenever all labels have that particular index.

Reply:
Thanks!